src/dmd/dtemplate.d

```diff
@@ -8288,46 +8288,108 @@ struct TemplateStats
     uint numInstantiations;     // number of instantiations of the template
     uint uniqueInstantiations;  // number of unique instantiations of the template
 
+    TemplateInstances* uniqueInstances;
+
     /*******************************
      * Add this instance
      */
-    static void incInstance(const TemplateDeclaration td)
+    static void incInstance(const TemplateDeclaration td,
+                            const TemplateInstance ti)
     {
         if (!global.params.vtemplates)
             return;
         if (!td)
             return;
+        if (!ti)
+            return;
         if (auto ts = cast(const void*) td in stats)
+        {
             ++ts.numInstantiations;
+        }
         else
+        {
             stats[cast(const void*) td] = TemplateStats(1, 0);
+        }
     }
 
     /*******************************
      * Add this unique instance
      */
-    static void incUnique(const TemplateDeclaration td)
+    static void incUnique(const TemplateDeclaration td,
+                          const TemplateInstance ti)
     {
         if (!global.params.vtemplates)
             return;
         if (!td)
             return;
+        if (!ti)
+            return;
         if (auto ts = cast(const void*) td in stats)
+        {
+            if (ts.uniqueInstances is null)
+            {
+                ts.uniqueInstances = new TemplateInstances();
+            }
+            ts.uniqueInstances.push(cast() ti);
             ++ts.uniqueInstantiations;
+        }
         else
+        {
             stats[cast(const void*) td] = TemplateStats(0, 1);
+        }
     }
 }
 
-
 void printTemplateStats()
 {
-    if (!global.params.vtemplates)
-        return;
+    static struct TemplateDeclarationStats
+    {
+        TemplateDeclaration td;
+        TemplateStats ts;
+        static int compare(scope const TemplateDeclarationStats* a,
+                           scope const TemplateDeclarationStats* b) nothrow @nogc pure
+        {
+            if (a.ts.uniqueInstantiations > b.ts.uniqueInstantiations)
+                return -1;
+            else if (a.ts.uniqueInstantiations == b.ts.uniqueInstantiations)
+                return 0;
+            else
+                return 1;
```

Comment:
Think you could rewrite this as

```suggestion
            return a.ts.uniqueInstantiations - b.ts.uniqueInstantiations;
```

Reply:
Rather

`return b.ts.uniqueInstantiations - a.ts.uniqueInstantiations;`